mysql_fdw.c

```diff
@@ -942,11 +950,15 @@ mysqlGetForeignPlan(PlannerInfo *root,RelOptInfo *baserel, Oid foreigntableid, F
 	 * field of the finished plan node; we can't keep them in private state
 	 * because then they wouldn't be subject to later planner processing.
 	 */
-	return make_foreignscan(tlist,
-							local_exprs,
-							scan_relid,
-							params_list,
-							fdw_private);
+	return make_foreignscan(tlist
```

Comment:
I tested without the patch, the crash is not reproducible.

Can the author of the patch please look into this issue...I will do further
testing on import foreign schema once this issue is fixed...



On Sat, Jul 4, 2015 at 1:28 PM, Ahsan Hadi <ahsan.hadi@enterprisedb.com>
wrote:

> Thanks.
>
> I was able to apply your patch and get the import foreign schema working
> successfully. However I am running into crash when i try to select from any
> of the mysql foreign table.
>
> It seem to be crashing in the external library. Is anyone else seeing this
> issue?
>
> postgres=# select * from test.emp;
> server closed the connection unexpectedly
> This probably means the server terminated abnormally
> before or while processing the request.
> The connection to the server was lost. Attempting reset: Failed.
>
> (gdb) bt
> #0  0x00007f8243a7c037 in raise () from /lib/x86_64-linux-gnu/libc.so.6
> #1  0x00007f8243a7f698 in abort () from /lib/x86_64-linux-gnu/libc.so.6
> #2  0x00000000007d7327 in ExceptionalCondition ()
> #3  0x000000000061ad47 in list_delete ()
> #4  0x00007f82399c31d4 in mysql_stmt_close (stmt=0x1d07de0)
>     at
> /export/home/pb2/build/sb_0-12116699-1399374319.34/mysql-5.6.19/libmysql/libmysql.c:4705
> #5  0x00007f8239f5d38a in mysqlEndForeignScan () from
> /home/edb/Projects/PG_servers/postgres/binary/lib/mysql_fdw.so
> #6  0x00000000005ffa5e in ExecEndForeignScan ()
> #7  0x00000000005db9d6 in standard_ExecutorEnd ()
> #8  0x00000000005a5203 in PortalCleanup ()
> #9  0x00000000007fbd13 in PortalDrop ()
> #10 0x00000000006e895c in PostgresMain ()
> #11 0x00000000004654b3 in ServerLoop ()
> #12 0x000000000068a6de in PostmasterMain ()
> #13 0x00000000004660c6 in main ()
> (gdb)
>
>
> -- Ahsan
>
> On Thu, Jul 2, 2015 at 9:55 PM, jjthiessen <notifications@github.com>
> wrote:
>
>> In mysql_fdw.c
>> <https://github.com/EnterpriseDB/mysql_fdw/pull/62#discussion_r33798674>:
>>
>> > @@ -942,11 +950,15 @@ mysqlGetForeignPlan(PlannerInfo *root,RelOptInfo *baserel, Oid foreigntableid, F
>> >  	 * field of the finished plan node; we can't keep them in private state
>> >  	 * because then they wouldn't be subject to later planner processing.
>> >  	 */
>> > -	return make_foreignscan(tlist,
>> > -							local_exprs,
>> > -							scan_relid,
>> > -							params_list,
>> > -							fdw_private);
>> > +	return make_foreignscan(tlist
>>
>> This branch is based on my PG_95 branch which should have fixed the
>> compile time errors that you listed (please see commit 5f8e57c
>> <https://github.com/EnterpriseDB/mysql_fdw/commit/5f8e57cf6324c0f1926c59ce034ca0f189804378>).
>> What branch are you on, and is 5f8e57c
>> <https://github.com/EnterpriseDB/mysql_fdw/commit/5f8e57cf6324c0f1926c59ce034ca0f189804378>
>> in the commit log?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/EnterpriseDB/mysql_fdw/pull/62/files#r33798674>.
>>
>
>
>
> --
> Ahsan Hadi
> Snr Director Product Development
> EnterpriseDB Corporation
> The Enterprise Postgres Company
>
> Phone: +92-51-8358874
> Mobile: +92-333-5162114
>
> Website: www.enterprisedb.com
> EnterpriseDB Blog: http://blogs.enterprisedb.com/
> Follow us on Twitter: http://www.twitter.com/enterprisedb
>
> This e-mail message (and any attachment) is intended for the use of the
> individual or entity to whom it is addressed. This message contains
> information from EnterpriseDB Corporation that may be privileged,
> confidential, or exempt from disclosure under applicable law. If you are
> not the intended recipient or authorized to receive this for the intended
> recipient, any use, dissemination, distribution, retention, archiving, or
> copying of this communication is strictly prohibited. If you have received
> this e-mail in error, please notify the sender immediately by reply e-mail
> and delete this message.
>



-- 
Ahsan Hadi
Snr Director Product Development
EnterpriseDB Corporation
The Enterprise Postgres Company

Phone: +92-51-8358874
Mobile: +92-333-5162114

Website: www.enterprisedb.com
EnterpriseDB Blog: http://blogs.enterprisedb.com/
Follow us on Twitter: http://www.twitter.com/enterprisedb

This e-mail message (and any attachment) is intended for the use of the
individual or entity to whom it is addressed. This message contains
information from EnterpriseDB Corporation that may be privileged,
confidential, or exempt from disclosure under applicable law. If you are
not the intended recipient or authorized to receive this for the intended
recipient, any use, dissemination, distribution, retention, archiving, or
copying of this communication is strictly prohibited. If you have received
this e-mail in error, please notify the sender immediately by reply e-mail
and delete this message.